HHS/OPRE-OPS
# Ops 674 put bli

## What changed

Update (PUT) BudgetLineItem.
Refactoring.

## Issue

https://app.zenhub.com/workspaces/opre-ops-62d197b4c468970013bcd2ec/issues/gh/hhs/opre-ops/674

## How to test

- Create a new BLI by POSTing to: `http://localhost:8080/api/v1/budget-line-items/` with the Request:

```
{
    "line_description": "LI 1",
    "comments": "hah hah",
    "agreement_id": 1,
    "can_id": 1,
    "amount": 200.24,
    "status": "DRAFT",
    "date_needed": "2024-01-01",
    "psc_fee_amount": 2.34
}
```

- You should receive a Response like:

```
{
    "agreement_id": 1,
    "amount": 200.24,
    "can_id": 1,
    "comments": "hah hah",
    "created_by": 12,
    "created_on": "2023-04-19T19:59:15.835202",
    "date_needed": "2024-01-01",
    "id": 20,
    "line_description": "LI 1",
    "psc_fee_amount": 2.34,
    "status": "DRAFT",
    "updated_on": "2023-04-19T19:59:15.835202"
}
```

Note the new BLI id - in this case 20

- Update BLI 21 by sending a PUT to: `http://localhost:8080/api/v1/budget-line-items/20` with the Request:

```
{
    "line_description": "LI 1",
    "comments": "hah hah",
    "agreement_id": 2,
    "can_id": 1,
    "amount": 200.24,
    "status": "DRAFT",
    "date_needed": "2024-01-01",
    "psc_fee_amount": 2.34
}
```

Note we're updating the `agreement_id` to 2

- You should receive a Response like:

```
{
    "agreement_id": 2,
    "amount": 200.24,
    "can_id": 1,
    "comments": "hah hah",
    "created_by": 12,
    "created_on": "2023-04-19T19:59:15.835202",
    "date_needed": "2024-01-01",
    "id": 20,
    "line_description": "LI 1",
    "psc_fee_amount": 2.34,
    "status": "DRAFT",
    "updated_on": "2023-04-19T20:00:20.366964"
}
```
